apps/hash-frontend/src/pages/drafts.page/draft-entity/draft-entity-action-buttons.tsx

```diff
@@ -122,10 +123,34 @@ export const DraftEntityActionButtons: FunctionComponent<{
     [outgoingDraftLinks, incomingDraftLinks],
   );
 
+  const { notifications, archiveNotification } = useNotifications();
+
+  const archiveRelatedNotifications = useCallback(async () => {
+    const relatedNotifications = notifications?.filter(
+      (notification) =>
+        notification.occurredInEntity.metadata.recordId.entityId ===
+        entity.metadata.recordId.entityId,
+    );
+
+    if (!relatedNotifications) {
+      return;
+    }
+
+    await Promise.all(
+      relatedNotifications.map((notification) => {
+        return archiveNotification({
+          notificationEntityId: notification.entity.metadata.recordId.entityId,
+        });
+      }),
+    );
+  }, [entity, notifications, archiveNotification]);
```

Comment:
Presumably this will need to happen on the entity editor page too, once that has a 'discard' button – maybe worth putting it somewhere shared now? We'll probably end up putting it more places too tbh (e.g. the browser plugin), so I expect we'll end up wanting to do it in a hook, but this works for now

Reply:
Hmm fair point, this isn't a solution for the browser plugin but I think I should just add `discardDraftEntity`/`acceptDraftEntity` functions to the draft entities context, so we can directly call a refetch within the context as well.

Reply:
I've added the methods in https://github.com/hashintel/hash/pull/3727/commits/10f6a8bdad632f5b68923515e2deea8c5ee05512, I've not added the logic related to the linked draft entities as it's tied very strongly to the modal but will probably make this re-usable in the entity editor PR